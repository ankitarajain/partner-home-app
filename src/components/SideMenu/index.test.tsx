import { render } from "@testing-library/react";
import { SideMenu } from "./index";

const defaultProps = {
  open: false,
  onClick: () => {},
  onItemSelected: () => {},
  data: [
    { id: "1", title: "1", hasAlert: true },
    {
      id: "2",
      title: "2",
      hasAlert: false,
      children: [{ id: "3", title: "3", hasAlert: true }],
    },
  ],
  selected: { id: "1", title: "1", hasAlert: true },
};
test("renders <SideMenu />", () => {
  expect(render(<SideMenu {...defaultProps} />)).toMatchSnapshot();
});
