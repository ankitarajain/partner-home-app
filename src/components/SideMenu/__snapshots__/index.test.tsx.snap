// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <SideMenu /> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiDrawer-root MuiDrawer-docked css-14vd0f8-MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
        >
          <div
            class="sc-dkPtRN eLtutk"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="sc-gsDKAQ hVQLqf"
          >
            <ul
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            >
              <div
                class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                style="margin-left: 0px;"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  style="color: purple;"
                >
                  <span
                    class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <span
                      class="MuiBadge-colorPrimary MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRightCircular css-6j13o8-MuiBadge-badge"
                    />
                  </span>
                  <div
                    class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    style="padding-left: 20px;"
                  >
                    1
                  </div>
                </li>
              </div>
              <div
                class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                style="margin-left: 0px;"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                  style="color: black;"
                >
                  <span
                    class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <span
                      class="MuiBadge-colorPrimary MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRightCircular MuiBadge-invisible css-eyhg5l-MuiBadge-badge"
                    />
                  </span>
                  <div
                    class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    style="padding-left: 20px;"
                  >
                    2
                  </div>
                  <div
                    class="sc-bdvvtL jQKEbI"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                  </div>
                </li>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiDrawer-root MuiDrawer-docked css-14vd0f8-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="sc-dkPtRN eLtutk"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="sc-gsDKAQ hVQLqf"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <div
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              style="margin-left: 0px;"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                style="color: purple;"
              >
                <span
                  class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <span
                    class="MuiBadge-colorPrimary MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRightCircular css-6j13o8-MuiBadge-badge"
                  />
                </span>
                <div
                  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  style="padding-left: 20px;"
                >
                  1
                </div>
              </li>
            </div>
            <div
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              style="margin-left: 0px;"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
                style="color: black;"
              >
                <span
                  class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <span
                    class="MuiBadge-colorPrimary MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRightCircular MuiBadge-invisible css-eyhg5l-MuiBadge-badge"
                  />
                </span>
                <div
                  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  style="padding-left: 20px;"
                >
                  2
                </div>
                <div
                  class="sc-bdvvtL jQKEbI"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                </div>
              </li>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
